crates/core/tedge_api/src/alarm.rs

```diff
@@ -1,18 +1,24 @@
 use std::convert::TryFrom;
 use std::convert::TryInto;
+use std::fmt;
 
 use clock::Timestamp;
 use serde::Deserialize;
+use serde::Serialize;
 
+use serde_json::Value;
+use std::collections::HashMap;
 /// In-memory representation of ThinEdge JSON alarm.
-#[derive(Debug, Deserialize, Eq, PartialEq)]
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
 pub struct ThinEdgeAlarm {
```

Comment:
I agree that it's slightly unusual. But, it was done this way because some of the alarm metadata like the `name` and `severity` are derived from the MQTT topic and the rest from the payload. The `ThinEdgeAlarmData` struct is used to parse the payload. We just reuse the same struct here by flattening it, instead of repeating the same fields in this struct.

Reply:
One should not design a struct from its external representation.